src/transformStateWithClones.js

```diff
@@ -6,8 +6,42 @@
  *
  * @return {Object[]}
  */
+const addProperties = 'addProperties';
+const removeProperties = 'removeProperties';
+const clear = 'clear';
+
 function transformStateWithClones(state, actions) {
-  // write code here
+  const STATES = [];
+  let CURRENT_STATE = { ...state };
+
+  for (const action of actions) {
+    let NEXT_STATE;
```

Comment:
The NEXT_STATE variable is unnecessary, you can change your current state directly.

Reply:
done